### ISSUE TITLE
DKIM body hash with fails with trailing line with space

### ISSUE THREAD

**Issue by nitinag** [jstedfast/MimeKit#187]
When signing a message with a body having a trailing line with only a WSP (space or tab) in it with body DkimCanonicalizationAlgorithm.Relaxed algorithm, DKIM verification fails as "body hash did not verify" (tested at gmail).

I'm guessing since it's at the end, the canonicalization algorithm is not removing what should now be an additional blank line at the end of the message with only a CRLF as part of step b. at https://tools.ietf.org/html/rfc6376#section-3.4.4.

```
message.Body = new TextPart("plain")
{
    Text = @"Hello!
        
"}
```
(Github won't show the extra space on the second line of the code example so you'll need to add it)

**Comment by jstedfast** [comment]
If you find any more of these, could you provide a unit test? It would make it easier to illustrate the issue.

It took me a while to figure out what you meant by whitespace at the end. I think I figured it out, but I'm still not 100% sure I understand you correctly.

**Comment by nitinag** [comment]
Sure, will try to do unit tests as I get more familiar with the library.

Just to provide some clarification, the body text from my example was: "Hello!\r\n  \r\n\r\n"
i.e. two spaces on the second line
That's what doesn't verify.

However, "Hello!\r\n  \r\nTest\r\n\r\n" verifies just fine.

**Comment by jstedfast** [comment]
I've released MimeKit 1.2.15 to nuget.org with a fix for this.
